### DISCUSSION TITLE
Install Github App from an Organization from another governing Github App

### DISCUSSION THREAD

**Discussion by marcellodesales** [community/community#107031] [Enterprise] [answered]
# 🤔 Distributed Github Apps

* We have a distributed environment, whose teams own automation through their own Github App
  * Each Github App has specific responsibilities around a GIthub org and its list of embedded repositories
  * Say GH App 1, GH App 2, GH App 3
  * A new governing Github App X was introduced

```
                ┌────────────┐
      ┌─────────┤  GH APP X  ├─────────┐
      │         └──────┬─────┘         │
      │                │               │
      │                │               │
┌─────▼──────┐  ┌──────▼─────┐  ┌──────▼─────┐
│  GH APP 1  │  │  GH APP 2  │  │  GH APP 3  │
└────────────┘  └────────────┘  └────────────┘
```

* We have a requirement that the governing Github App X needs to install Github APP 1, 2, 3 into a Github Org... 
  * However, without human intervention
  * This needs to be in the API level

# 👽 API Available

For what I understand from the docs, and ChatGPT response, is that I simply need to retrieve a an `access_token` from an Organization JWT token to an app... Maybe chatGPT hallucinated in regard to the param 3.10... https://docs.github.com/en/enterprise-server@3.10/rest/authentication/endpoints-available-for-github-app-installation-access-tokens?apiVersion=2022-11-28#apps but I can't see that.. 

* Question: is there a way to programmatically get GH App X install other GH Apps into a given `organization_Id`?

# 🧠 ChatGPT

* According to chatGPT: `
How to install a Github App to a Github Organization from another GIthub App? How can we automate that using curl apis?`

```
1. Generate JWT tokens: As per the [documentation](https://docs.github.com/en/authentication/getting-started-with-authentication/about-authentication#json-web-tokens), you'll need a JWT token with a private key for server-to-server communication.
2. Grant the necessary permissions: Ensure the necessary permissions are granted to your app in the "Permissions & Webhooks" section (such as organizations, administration). Now, proceed to install the app to the organization.
  * Use curl to install the app to the organization: After setting up your GitHub App, use the following curl command to install the app to the organization:

```console
curl -i -X PUT \
  -H "Authorization: Bearer <jwt_token>" \
  -H "Accept: application/vnd.github.machine-man-preview+json" \
  https://api.github.com/app/installations/<app_id>/access_tokens \
  -d '{
    "organization_id": <organization_id>
}'
```
> **NOTES**: 
> * Replace <jwt_token> with your generated JWT token, <app_id> with your GitHub App's ID, and <organization_id> with the ID of the organization where you want to install your GitHub App.
```

**Comment by hpsin** [comment, marked as answer]
Yep, that's a hallucination. There's no programmatic way to install an app using another app. It's a long term plan for us but there's nothing on the roadmap at the moment.

**Reply by marcellodesales** [reply]
@hpsin please ask the team to update this thread once that's in place... Meanwhile, is there any other alternative other than asking users to install 5-10 apps just because of the lack of this feature?
